Source/WebCore/rendering/RenderTableCell.cpp

```diff
@@ -1379,17 +1399,31 @@ void RenderTableCell::paintBackgroundsBehindCell(PaintInfo& paintInfo, const Lay
     if (backgroundObject != this)
         adjustedPaintOffset.moveBy(location());
 
+    // Background images attached to the row or row group must span the row
+    // or row group. Draw them at the backgroundObject's dimensions, but
+    // clipped to this cell.
+    // FIXME: This should also apply to columns and column groups.
+    bool paintBackgroundObject = backgroundObject != this && bgLayer.hasImage() && !is<RenderTableCol>(backgroundObject);
     // We have to clip here because the background would paint
     // on top of the borders otherwise. This only matters for cells and rows.
-    bool shouldClip = backgroundObject->hasLayer() && (backgroundObject == this || backgroundObject == parent()) && tableElt->collapseBorders();
+    bool shouldClip = paintBackgroundObject || (backgroundObject->hasLayer() && (backgroundObject == this || backgroundObject == parent()) && tableElt->collapseBorders());
     GraphicsContextStateSaver stateSaver(paintInfo.context(), shouldClip);
     if (shouldClip) {
         LayoutRect clipRect(adjustedPaintOffset.x() + borderLeft(), adjustedPaintOffset.y() + borderTop(),
             width() - borderLeft() - borderRight(), height() - borderTop() - borderBottom());
         paintInfo.context().clip(clipRect);
     }
+    LayoutRect fillRect;
+    if (paintBackgroundObject) {
+        if (auto* sectionElt = dynamicDowncast<RenderTableSection>(backgroundObject))
```

Comment:
what's sectionElt?

Reply:
Not sure what you mean? `sectionElt` is `backgroundObject`, in the case that that element is a RenderTableSection. If you are criticizing the name, please provide an alternative.

Reply:
Should there be a comment added to explain the logic?

Reply:
tableSectionRenderer may be a better candidate (1. this is a renderer and not an element, 2. [full words please](https://webkit.org/code-style-guidelines/#names-full-words))